### ISSUE TITLE
Retrieve system_profiler as XML so you have more info (also, it will be way easier to parse)

### ISSUE THREAD

**Issue by vitor251093** [sebhildebrandt/systeminformation#567]
**Is your feature request related to a problem? Please describe.**
At this moment, when retrieving the graphic card, you use the following command:
```
system_profiler SPDisplaysDataType
```
That command returns the result in a human readable way, which is not that good for parsing (I'm sure you already noticed that), and also omits some informations.

**Describe the solution you'd like**
Instead, you could use this command:
```
system_profiler -xml -detailLevel full SPDisplaysDataType
```
The output can be easily parsed with the [plist npm project](https://www.npmjs.com/package/plist), so you don't have to deal with parsing a human readable output in case Apple changes it some day (after all, that output was not mean to be parsed at all, so in theory, they can change without warning). You can also parse the XML manually with Node/JS if you prefer; it should still be much easier than parsing the human output.

**Describe alternatives you've considered**
At this moment, my project uses systeminformation, but I've started getting the video card information myself. I also had the need to retrieve the vendorId and deviceId of the graphic cards, which this method allows me to do with ease. 

With this method you can also, for example, check if the graphic card kext is loaded (the kext not being loaded is a common issue among Hackintosh users, since graphic cards are useless in macOS without the kext). You can also check if the graphic card has Metal support, and even how good that support is, if you want.

**Additional context**
Before working on that using Node, I've used to parse this output in Objective-C, and I've open sourced that part of the project. So in case you want to support the XML output, but don't know which key names to expect, you can just check here:
https://github.com/vitor251093/ObjectiveC_Extension/blob/master/ObjectiveC_Extension/VMMVideoCard.h
https://github.com/vitor251093/ObjectiveC_Extension/blob/master/ObjectiveC_Extension/VMMVideoCard.m

I know it doesn't include the keys for everything you need, but it should at least be a start. I would create a pull request myself, making the changes to systeminformation, but unfortunately I don't have enough free time for that in my hands.

Those two files also include support for retrieving the graphic cards data using `IOServiceMatching` and the OpenGL framework, but I haven't tried to implement those in Node yet, and I sincerely don't know if they would still be necessary in modern computers (back in 2019, the methods in this class combined could detect the graphic cards properly in around 99.9942% of the computers they were tested in; 522.556 samples, 30 failures), but I can't say for sure how much of those were aided by `IOServiceMatching` and the OpenGL framework. Also, considering that the OpenGL framework is walking on the deprecation route, maybe it would be better to just let it go.

**Comment by sebhildebrandt** [comment]
@vitor251093 thank you for your ideas! I am considering changing this to use JSON (instead of XML) which makes parsing even easier.

**Comment by sebhildebrandt** [comment]
@vitor251093 I managed to implement the XML parsing part (without dependencies) in < 100 lines. As I am quite busy for the next 7 days, I will try to finish this afterwards. Thank you for hint to use XML.

**Comment by sebhildebrandt** [comment]
@vitor251093 ... what are the properties you would like to see as additional properties in the graphics section?

**Comment by vitor251093** [comment]
- The video card chipset name (_name)
- The video card model name (sppci_model), which I assume you already mapped for that first version
- The video card device ID (spdisplays_device-id)
- The video card vendor ID (spdisplays_vendor-id)
- The video card vendor name (spdisplays_vendor)
- The video card bus type (sppci_bus)
- The video card memory size if has a PCI/PCIe bus type (spdisplays_vram)
- The video card memory size if it's a built-in card (spdisplays_vram_shared)
- The video card Kext info if available (sppci_kextinfo)
- The video card device type (sppci_device_type), which is useful to determine if it's an external video card (value spdisplays_egpu)
- The video card supported Metal version (spdisplays_metal)

**Comment by sebhildebrandt** [comment]
@vitor251093 so I now replaced the old paring method by my new XML parser. Should now return the same results as the previous version plus added properties:

**For  Graphic Devices**
- deviceId
- vendorId
- external 
- cores (M1 hardware only)
- metalVersion

**For Displays**
- vendorId
- productionYear
- serial
- displayId

Can you test this on your side (not yet published on NPM but current code here on gitHub should reflect the changes). Any further ideas to that?

**Comment by vitor251093** [comment]
Well, I do have one more suggestion related to this topic, but it's more of a workaround for a common problem. So it may be helpful for people, but it's still a workaround. I can illustrate the problem with my controllers output:

```
controllers: [{
    "bus": "Built-In",
    "cores": null,
    "deviceId": "0x0a2e",
    "external": false,
    "metalVersion": "mac1",
    "model": "Intel Iris",
    "vendor": "Intel",
    "vendorId": "",
    "vram": 1536,
    "vramDynamic": true
}]
```
As you can see, vendorId is not available. That happens in some computers, and I really don't know why. Still, since `vendor` is available, that value can be determined. I use a method like that one to get the vendor ID myself:

```
            var localVendorID = dictionary["spdisplays_vendor-id"];
            if (localVendorID != undefined) {
                return localVendorID.toLowerCase()
            }
            
            var localVendor = dictionary["spdisplays_vendor"]; // eg. 'NVIDIA (0x10de)' or 'sppci_vendor_Nvidia'
            if (localVendor == undefined) {
                return null
            }
            
            if (localVendor.includes("(")) {
                return localVendor.split("(").pop().split(")").shift()
            }
            
            if (localVendor.startsWith("sppci_vendor_")) {
                localVendor = localVendor.replace("sppci_vendor_", "")
                localVendor = localVendor.toUpperCase();
                
                if (localVendor.includes("APPLE")) {
                    return "0x05ac"
                }
                if (localVendor.includes("NVIDIA")) {
                    return "0x10de"
                }
                if (localVendor.includes("ATI") || localVendor.includes("AMD")) {
                    return "0x1002"
                }
                if (localVendor.includes("INTEL")) {
                    return "0x8086"
                }
            }
            
            return null
```

This method gave me no problems until today, but that's up to you to decide.

Also, in a "kinda" unrelated topic: I see `virtual` from `si.system` is unimplemented for macOS according to the README file. It may not be a fail proof method, but you can use the video card to determine if the user is running under a VM, since VMs show "fake video cards" instead of the host computer one, unless the user do something to specifically workaround that. If the vendor ID of the video card is only of the following, you are 100% on a VM:

- QEMU: "0x1234"
- VMWare: "0x15ad"
- VirtualBox: "0x80ee"
- Parallels Desktop: "0x1ab8"
- Microsoft Remote Desktop: "0xbaad"

**Comment by sebhildebrandt** [comment]
@vitor251093 I implemented a version based on your data to provide a vendor ID if this is empty. Version 5.9.0 just published. 

For the `virtual` issue: this is an awesome idea. Will add this soon.
Closing this for now.

**Comment by sebhildebrandt** [comment]
@vitor251093 just published a new version (5.9.2) with improved XML parsing (I had some minor issues on 2 of my macOS machines). If you discover more issues just feel free to comment here.
